src/components/info-tag/InfoTag.tsx

```diff
@@ -2,16 +2,26 @@ import {StyleProp, ViewStyle} from 'react-native';
 import {StyleSheet} from '@atb/theme';
 import React from 'react';
 import {InfoChip} from '../info-chip';
+import {InfoTagTexts, useTranslation} from '@atb/translations';
 
 type Props = {
-  text: string;
+  mode: 'beta' | 'new';
   style?: StyleProp<ViewStyle>;
 };
-export const InfoTag = ({text, style}: Props) => {
+export const InfoTag = ({mode, style}: Props) => {
   const styles = useStyles();
+  const {t} = useTranslation();
+
+  const getLocalizedText = () => {
+    if (mode === 'new') return t(InfoTagTexts.newText);
+    else {
+      return 'Beta';
+    }
+  };
```

Comment:
Not something that needs to be fixed, just a FYI:

Creating the `getLocalizedText` function like this is not preferred, as the function is recreated every render for no real reason. It would be more efficient, and cleaner, to write this as:
```tsx
const text = mode === 'new' ? t(InfoTagTexts.newText) : 'Beta';
```
However, when handling union values like this (`'beta' | 'new'`) it is preferred to use `switch` statement, as it gives more type-safety when adding new `mode` values. But a switch statement doesn't really look good inline in the render function, so then it is appropriate to create a function for it. Note then that it is preferred to create it outside of the render function, so it isn't recreated on every render. Something like this:
```tsx
const getLocalizedText = (mode: Props['mode']) => {
  switch (mode) {
    case 'new':
      return t(InfoTagTexts.newText);
    case 'beta':
      return 'Beta';
  }
}
```

Doing it like this will give typescript error during compiling if a new `mode` is added, without being handled by the `switch` statement. Just doing `if`/`else` will not give that type of compiler help.

Reply:
Yes that definitely makes sense💡